src/lib.rs

```diff
@@ -148,30 +236,27 @@ impl UPClientZenoh {
     fn attachment_to_uattributes(attachment: &Attachment) -> anyhow::Result<UAttributes> {
         let mut attachment_iter = attachment.iter();
         if let Some((_, value)) = attachment_iter.next() {
-            let version = *value.as_slice().first().ok_or(UStatus::fail_with_code(
-                UCode::INTERNAL,
-                "uAttributes version is empty",
-            ))?;
+            let version = *value.as_slice().first().ok_or_else(|| {
+                let msg = "UAttributes version is empty".to_string();
```

Comment:
WDYT about adding some identifying information as to which message had these failures, e.g. logging at a minimum the `UMessage.attributes.id`?

I think it would help traceability for an end-user

Reply:
Good idea. I'll add it